sdk/core/System.ClientModel/src/Pipeline/ApiKeyAuthenticationPolicy.cs

```diff
@@ -0,0 +1,129 @@
+    {
+        // TODO: Add tests for this implementation if the API is approved.
+        Argument.AssertNotNull(credential, nameof(credential));
+        Argument.AssertNotNullOrEmpty(queryName, nameof(queryName));
+
+        return new ApiKeyAuthenticationPolicy(credential, queryName, KeyLocation.Query);
+    }
+
+    public static ApiKeyAuthenticationPolicy CreateBasicAuthorizationPolicy(ApiKeyCredential credential)
+    {
+        Argument.AssertNotNull(credential, nameof(credential));
+
+        return new ApiKeyAuthenticationPolicy(credential, "Authorization", KeyLocation.Header, "Basic");
+    }
+
+    public static ApiKeyAuthenticationPolicy CreateBearerAuthorizationPolicy(ApiKeyCredential credential)
+    {
+        Argument.AssertNotNull(credential, nameof(credential));
+
+        return new ApiKeyAuthenticationPolicy(credential, "Authorization", KeyLocation.Header, "Bearer");
+    }
+
+    private ApiKeyAuthenticationPolicy(ApiKeyCredential credential, string name, KeyLocation keyLocation, string? keyPrefix = null)
+    {
+        Argument.AssertNotNull(credential, nameof(credential));
+        Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+        _credential = credential;
+
+        _name = name;
+        _location = keyLocation;
+        _keyPrefix = keyPrefix;
+    }
+
+    public sealed override void Process(PipelineMessage message, IReadOnlyList<PipelinePolicy> pipeline, int currentIndex)
+    {
+        SetKey(message);
+
+        ProcessNext(message, pipeline, currentIndex);
+    }
+
+    public sealed override async ValueTask ProcessAsync(PipelineMessage message, IReadOnlyList<PipelinePolicy> pipeline, int currentIndex)
+    {
+        SetKey(message);
+
+        await ProcessNextAsync(message, pipeline, currentIndex).ConfigureAwait(false);
+    }
+
+    private void SetKey(PipelineMessage message)
+    {
+        switch (_location)
+        {
+            case KeyLocation.Header:
+                SetHeader(message);
+                break;
+            case KeyLocation.Query:
+                AddQueryParameter(message);
+                break;
+            default:
+                throw new InvalidOperationException($"Unsupported value for Key location: '{_location}'.");
+        }
+    }
+
+    private void SetHeader(PipelineMessage message)
+    {
+        _credential.Deconstruct(out string key);
+
+        message.Request.Headers.Set(_name, _keyPrefix != null ? $"{_keyPrefix} {key}" : key);
+    }
+
+    private void AddQueryParameter(PipelineMessage message)
+    {
+        _credential.Deconstruct(out string key);
+
+        StringBuilder query = new StringBuilder();
+        query.Append(_name);
+        query.Append('=');
+        query.Append(Uri.EscapeDataString(key));
+
+        UriBuilder uriBuilder = new(message.Request.Uri);
+        uriBuilder.Query = (uriBuilder.Query != null && uriBuilder.Query.Length > 1) ?
```

Comment:
I fear this might have bugs (duplicate or override query key-value pairs). I propose we remove query support from v1.

Reply:
Tracking adding it back in https://github.com/Azure/azure-sdk-for-net/issues/41391